From: Hendrik Brueckner <brueckner@redhat.com>
Date: Fri, 11 Dec 2009 10:07:44 -0500
Subject: [s390] kernel: fix dump indicator
Message-id: <20091211100744.GB4579@redhat.com>
Patchwork-id: 21877
O-Subject: [RHEL5 U5 PATCH 1/1] s390 - kernel: fix dump indicator
Bugzilla: 546288
RH-Acked-by: Pete Zaitcev <zaitcev@redhat.com>
RH-Acked-by: Jarod Wilson <jarod@redhat.com>

Description
-----------
Kernel dumps are not readable by the lcrash or crash utility.
The problem is that in certain cases, an incorrect dump might
be written.
Set dump indicator on read-scp-info command to get meaningful
dumps.
This patch provides the RHEL 5 backport of the upstream version.

Bugzilla
--------
BZ 546288
https://bugzilla.redhat.com/show_bug.cgi?id=546288

Upstream status of the patch
----------------------------
The patch will be upstream as of kernel version 2.6.33
http://git.kernel.org/?p=linux/kernel/git/torvalds/linux-2.6.git;a=commitdiff;h=cb9d71604a6e3d6aa93bb663747a62fc520da483

Test status
-----------
The patch has been tested and fixes the problem.
The fix has been verified by the IBM test department.

Please ACK.

With best regards,

	Hendrik


diff --git a/arch/s390/kernel/head31.S b/arch/s390/kernel/head31.S
index e2b3c14..b227b37 100644
--- a/arch/s390/kernel/head31.S
+++ b/arch/s390/kernel/head31.S
@@ -295,7 +295,7 @@ ipl_schib:
 s390_readinfo_sccb:
 .Lsccb:
 	.hword	0x1000			# length, one page
-	.byte	0x00,0x00,0x00
+	.byte	0x80,0x00,0x00
 	.byte	0x80			# variable response bit set
 .Lsccbr:
 	.hword	0x00			# response code
diff --git a/arch/s390/kernel/head64.S b/arch/s390/kernel/head64.S
index c062898..5446d81 100644
--- a/arch/s390/kernel/head64.S
+++ b/arch/s390/kernel/head64.S
@@ -371,7 +371,7 @@ ipl_schib:
 s390_readinfo_sccb:
 .Lsccb:
 	.hword 0x1000			# length, one page
-	.byte 0x00,0x00,0x00
+	.byte 0x80,0x00,0x00
 	.byte 0x80			# variable response bit set
 .Lsccbr:
 	.hword 0x00			# response code
